core/src/test/scala/unit/kafka/server/AbstractApiVersionsRequestTest.scala

```diff
@@ -85,6 +85,9 @@ abstract class AbstractApiVersionsRequestTest(cluster: ClusterInstance) {
 
       assertEquals(0, apiVersionsResponse.data().supportedFeatures().find(EligibleLeaderReplicasVersion.FEATURE_NAME).minVersion())
       assertEquals(EligibleLeaderReplicasVersion.ELRV_1.featureLevel(), apiVersionsResponse.data().supportedFeatures().find(EligibleLeaderReplicasVersion.FEATURE_NAME).maxVersion())
+
+      assertEquals(0, apiVersionsResponse.data().supportedFeatures().find(ShareVersion.FEATURE_NAME).minVersion())
+      assertEquals(ShareVersion.SV_1.featureLevel(), apiVersionsResponse.data().supportedFeatures().find(ShareVersion.FEATURE_NAME).maxVersion())
```

Comment:
Hmm, this is only correct if unstable.feature.versions.enable is set to true. However, some of the callers like SaslApiVersionsRequestTest doesn't seem to set unstable.feature.versions.enable to true. So, how does this work?

Reply:
The aim here is that `unstable.feature.versions.enable=true` should not be required to enable share groups in AK 4.1. They should be disabled on new brokers by default, but possible to enable using `kafka-features.sh upgrade`. I'm trying to do what was done for the ELR preview (https://github.com/apache/kafka/pull/17867 and then https://github.com/apache/kafka/pull/18954).

In AK 4.2, the metadata version will be `IBP_4_2_IV0`, so a new broker should get `share.version=1` by default.